### PR TITLE
Froge example

### PR DESCRIPTION
**Pull request by lvd0**

### CODE REVIEW

**Review by deccer — CHANGES_REQUESTED** [review]
i think we should have it all in ProjectApplication, no abstractions like Shader/Mesh/Model, since this is an absolute starter thing. thats up to the peeps trying to use this as a starter.

But the readme could have a word about it, that the code could be abstracted into Project.Library/Mesh/Model/Shader along with what needs to be massaged in CMakeLists for it to happen

**Review by deccer — CHANGES_REQUESTED** [review]
```
lib/cgltf.c
lib/stb_image.c
```
should not be part of this either, since both are pulled by CMake

File: src/Project/Model.cpp [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    {
+        const auto& material = model->materials[i];
+        const auto* image = material.pbr_metallic_roughness.base_color_texture.texture->image;
+        const auto texturePath = FindTexturePath(basePath, image);
+        if (textureIds.contains(texturePath))
+        {
+            continue;
+        }
+        uint32_t texture;
+        glGenTextures(1, &texture);
```
Comment (deccer) [inline src/Project/Model.cpp]: heh, you also use non DSA versions here, any reason?

File: src/Project/Model.cpp [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+        glBindTexture(GL_TEXTURE_2D, texture);
+
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT);
```
Comment (deccer) [inline src/Project/Model.cpp]: also non DSA variant

File: src/Project/Model.cpp [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+    }
+    if (!glMakeTextureHandleResidentARB)
+    {
+        glMakeTextureHandleResidentARB = (void(*)(GLuint64))glfwGetProcAddress("glMakeTextureHandleResidentARB");
+    }
+    // Allocate GL buffers
+    glCreateVertexArrays(1, &_vao);
+    glCreateBuffers(1, &_vbo);
+    glCreateBuffers(1, &_ibo);
+    glGenBuffers(1, &_objectData);
```
Comment (deccer) [inline src/Project/Model.cpp]: non DSA variant here, any reason?

File: src/Project/Model.cpp [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+    else
+    {
+        texturePath = (basePath / image->uri).generic_string();
+    }
+    return texturePath;
+}
+
+Model::Model(std::string_view file)
+{
+    if (!glGetTextureHandleARB)
```
Comment (deccer) [inline src/Project/Model.cpp]: this should be covered by glad out of the box, i dont think checking for null is necessary, we are requesting a gl4.6 context

**Review by deccer — CHANGES_REQUESTED** [review]

File: src/Project.Library/Application.cpp [hunk 5]
```diff
@@ -91,6 +92,26 @@ bool Application::Initialize()
...
+                              GLenum type,
+                              GLuint,
+                              GLenum severity,
+                              GLsizei,
+                              const GLchar* message,
+                              const void*)
+    {
+        if (type == GL_DEBUG_TYPE_ERROR)
+        {
+            std::printf(
```
Comment (deccer) [inline src/Project.Library/Application.cpp]: spdlog::error :)

File: src/Project.Library/Application.cpp [hunk 6]
```diff
@@ -56,8 +56,9 @@ bool Application::Initialize()
     glfwWindowHint(GLFW_DECORATED, GLFW_TRUE);
     glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_API);
     glfwWindowHint(GLFW_RESIZABLE, GLFW_TRUE);
-    glfwWindowHint(GLFW_VERSION_MAJOR, 4);
-    glfwWindowHint(GLFW_VERSION_MINOR, 6);
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
```
Comment (deccer) [inline src/Project.Library/Application.cpp:59]: good find :P

File: src/Project/Model.cpp (hunk 4 above)
Comment (lvd0) [reply src/Project/Model.cpp]: Unfortunately, there are no such functions in `glad.h`, so I had to load them manually.
I think they should be present in `glad.h` if you generate it enabling the required extensions

**Review by deccer — APPROVED** [review]
Looks good now

**Pull request merged**